lib/classes/hint.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:math';
+
+import 'package:phone_factory/classes/Ship.dart';
+import 'package:phone_factory/classes/bay_item.dart';
+
+class Hint {
+  int _currentBayNumber;
+  Ship ship = Ship();
+
+  Hint({required int currentBayNumber}) : _currentBayNumber = currentBayNumber;
+
+  void getHint() {
+    Map<int, dynamic> baysData = ship.getSpaceShipBays();
+    List<int> alloweBaysList = List.from(baysData.keys);
+    alloweBaysList.remove(_currentBayNumber);
```

Comment:
Move this logic outside this class. Make single responsibility for take alloweBaysList and give the hint for HintGenerator class. All other logic needs to be placed separate. You can create `alloweBaysList` getter inside Ship class